File: src/com/ksn/selection/SelectionSort.java
package com.ksn.selection;

import java.util.Arrays;
import java.util.stream.IntStream;

public class SelectionSort {

    public void sort(int [] arr){
        int n = arr.length;

        // One by one move boundary of unsorted subarray
        for (int i = 0; i < n-1; i++)
        {
            // Find the minimum element in unsorted array
            int min_idx = i;
            for (int j = i+1; j < n; j++)
                if (arr[j] < arr[min_idx])
                    min_idx = j;

            // Swap the found minimum element with the first
            // element
            int temp = arr[min_idx];
            arr[min_idx] = arr[i];
            arr[i] = temp;
        }
    }

    public static void main(String ...s) {

        SelectionSort selectionSort = new SelectionSort();

        int arr[] = {3,2,6,5,8,923,6,34,12,65,3};
        selectionSort.sort(arr);

        IntStream intStream = Arrays.stream(arr);
        intStream.forEach(x -> System.out.println(x));

    }
}

File: src/com/ksn/selection/selection-sort.py

class SelectionSort:
    def __init__(self):
        pass

    def sort(self, arr):
        n = len(arr)
        #One by one move boundary of unsorted subarray
        for i in range(0, n-1):
            #Find the minimum element in unsorted array
            min_idx = i
            for j in range(i+1, n):
                if arr[j] < arr[min_idx]:
                    min_idx = j

            #Swap the found minimum element with the first elem
            temp = arr[min_idx]
            arr[min_idx] = arr[i]
            arr[i] = temp


selectionSort = SelectionSort()
arr = [3,4,6,3,7,89,23,12,54]
selectionSort.sort(arr)
print(arr)